### ISSUE TITLE
Consider setting custom labels on managed resources

### ISSUE THREAD

**Issue by christophd** [joyrex2001/kubedock#24]
Custom labels are translated to annotations on the managed resource (e.g. deployments, services). I wonder why not also set labels on the managed resource?

Some platforms make a difference between annotations and labels on resources. I think it would make sense to set both on kubedock managed resources.

**Comment by christophd** [comment]
BTW many thanks for the awesome work and kubedock!

**Comment by joyrex2001** [comment]
The reasoning of using annotations is the difference of how [docker defines labels](https://docs.docker.com/config/labels-custom-metadata/), and how [kubernetes defines labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set). The value that is set in the docker-label might not be compatible with the k8s label requirements. This would mean the value can't be adopted 1:1 in the label, and requires some sanitisation.

Is there a specific use case you have in mind? E.g. would you want to have all labels bluntly copied over, or do you want to be able to add a label to a container and get that specific label propagated?

**Comment by christophd** [comment]
Thank you for the clarification. My usecase would be to set labels such as `app.kubernetes.io/name` or `app.openshift.io/part-of` (see https://github.com/redhat-developer/app-labels/blob/master/labels-annotation-for-openshift.adoc). Platforms such as OpenShift use this label information to set icons, connections and groups in the UI (e.g. https://docs.openshift.com/container-platform/4.11/applications/odc-viewing-application-composition-using-topology-view.html#odc-grouping-multiple-components_viewing-application-composition-using-topology-view)

So being able to add labels that translate to labels on the managed resource would be sufficient for that use case. Either copy all labels or use some filtering to propagate specific labels would work.

**Comment by joyrex2001** [comment]
Are you able to populate them via the docker labels, or do you need another interface?

If docker-labels is an accepted interface, I would like to be able to control which labels will be propagated and which not. Not sure how that filter would look like though...

**Comment by christophd** [comment]
yes, I think the docker labels api interface would be totally sufficient.

**Comment by christophd** [comment]
awesome! many thanks! 

please let me know when this change is available (maybe in v0.9.1?) so I can consume this enhancement

I think we can close this issue as fixed

**Comment by joyrex2001** [comment]
Great :-) I released https://github.com/joyrex2001/kubedock/releases/tag/0.9.1 with this enhancement and another small bugfix.

**Comment by christophd** [comment]
works like a charm. many thanks!
